rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/CoordinatorBookKeeper.java

```diff
@@ -43,24 +52,91 @@
   protected Cache<String, Boolean> fileValidatedWorkerCache;
 
   private final boolean isValidationEnabled;
+  private static Integer lock = 1;
+  protected ClusterManager clusterManager;
+  int clusterType;
 
-  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoundException
+  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics) throws BookKeeperInitializationException
   {
     this(conf, metrics, Ticker.systemTicker());
   }
 
   @VisibleForTesting
-  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws FileNotFoundException
+  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws BookKeeperInitializationException
   {
     super(conf, metrics, ticker);
     this.isValidationEnabled = CacheConfig.isValidationEnabled(conf);
     this.liveWorkerCache = createHealthCache(conf, ticker);
     this.cachingValidatedWorkerCache = createHealthCache(conf, ticker);
     this.fileValidatedWorkerCache = createHealthCache(conf, ticker);
+    this.clusterType = CacheConfig.getClusterType(conf);
 
+    initializeCoordinator();
+  }
+
+  private void initializeCoordinator() throws CoordinatorInitializationException
+  {
+    initializeClusterManager();
+    setCurrentNodeName();
     registerMetrics();
   }
 
+  @VisibleForTesting
+  protected void initializeClusterManager() throws CoordinatorInitializationException
```

Comment:
This function is used in initializeCoordinator (line:79) which is not a test function. So, I think you can remove **VisibleForTesting** annotation

Reply:
The `@VisibleForTesting` annotation "indicates that the visibility of a type or member has been relaxed to make the code testable."  
`initializeCoordinator` is used in test classes and making it protected should work. `@VisibleForTesting` tag is just to make sure we document why the scope of this method is relaxed.